go/vt/vttablet/tabletserver/connpool/dbconn.go

```diff
@@ -324,34 +325,35 @@ func (dbc *DBConn) Close() {
 	dbc.conn.Close()
 }
 
-func (dbc *DBConn) ApplySettings(ctx context.Context, settings []string) error {
-	for _, q := range settings {
-		if _, err := dbc.execOnce(ctx, q, 1, false); err != nil {
-			return err
-		}
+// ApplySetting implements the pools.Resource interface.
+func (dbc *DBConn) ApplySetting(ctx context.Context, setting pools.Setting) error {
+	query := setting.GetQuery()
+	if _, err := dbc.execOnce(ctx, query, 1, false); err != nil {
+		return err
 	}
-	dbc.settings = settings
+	dbc.setting = query
+	dbc.resetSetting = setting.GetResetQuery()
 	return nil
 }
 
-func (dbc *DBConn) IsSettingsApplied() bool {
-	return len(dbc.settings) > 0
+// IsSettingApplied implements the pools.Resource interface.
+func (dbc *DBConn) IsSettingApplied() bool {
+	return dbc.setting != ""
 }
 
-func (dbc *DBConn) IsSameSetting(settings []string) bool {
-	return compareStringSlice(settings, dbc.settings)
+// IsSameSetting implements the pools.Resource interface.
+func (dbc *DBConn) IsSameSetting(setting string) bool {
+	return strings.EqualFold(setting, dbc.setting)
```

Comment:
What would happen if the settings are the same but reordered? Like `set x=a, y=b` and `set y=b, x=a`? Do we consider them to be different settings?

Reply:
that is on vtgate to send in same order, it will not be handled here. 
VTGate already does that today in sending in alphanumeric sorted.